usr/lib/cca_stdll/cca_specific.c

```diff
@@ -1090,6 +1092,231 @@ CK_RV token_specific_rsa_decrypt(STDLL_TokData_t * tokdata,
+    OBJECT *key_obj = NULL;
+    CK_RV rc;
+
+    UNUSED(tokdata);
+    UNUSED(hash);
+    UNUSED(hlen);
+
+    rc = object_mgr_find_in_map1(tokdata, ctx->key, &key_obj, READ_LOCK);
+    if (rc != CKR_OK) {
+        TRACE_DEVEL("object_mgr_find_in_map1 failed\n");
+        goto done;
+    }
+
+    /* Find the secure key token */
+    if (!template_attribute_find(key_obj->template, CKA_IBM_OPAQUE, &attr)) {
+        TRACE_ERROR("%s\n", ock_err(ERR_TEMPLATE_INCOMPLETE));
+        rc = CKR_TEMPLATE_INCOMPLETE;
+        goto done;
+    }
+
+    oaep = (CK_RSA_PKCS_OAEP_PARAMS *)ctx->mech.pParameter;
+    if (oaep == NULL ||
+        ctx->mech.ulParameterLen != sizeof(CK_RSA_PKCS_OAEP_PARAMS)) {
+        TRACE_ERROR("%s\n", ock_err(ERR_MECHANISM_PARAM_INVALID));
+        rc = CKR_MECHANISM_PARAM_INVALID;
+        goto done;
+    }
+
+    if (oaep->source == CKZ_DATA_SPECIFIED && oaep->ulSourceDataLen > 0) {
+        TRACE_ERROR("CCA does not support non-empty OAEP source data\n");
+        rc = CKR_MECHANISM_PARAM_INVALID;
+        goto done;
+    }
+
+    /* The max value allowable by CCA for out_data_len is 512, so cap the
+     * incoming value if its too large. CCA will throw error 8, 72 otherwise.
+     */
+    if (*out_data_len > 512)
+        *out_data_len = 512;
+
+    rule_array_count = 2;
+    switch (oaep->hashAlg) {
+    case CKM_SHA_1:
+        if (oaep->mgf != CKG_MGF1_SHA1) {
+            TRACE_ERROR("%s\n", ock_err(ERR_MECHANISM_PARAM_INVALID));
+            rc = CKR_MECHANISM_PARAM_INVALID;
+            goto done;
+        }
+        memcpy(rule_array, "PKCSOAEPSHA-1   ", 2 * CCA_KEYWORD_SIZE);
+        break;
+
+    case CKM_SHA256:
+        if (oaep->mgf != CKG_MGF1_SHA256) {
+            TRACE_ERROR("%s\n", ock_err(ERR_MECHANISM_PARAM_INVALID));
+            rc = CKR_MECHANISM_PARAM_INVALID;
+            goto done;
+        }
+        memcpy(rule_array, "PKCSOAEPSHA-256 ", 2 * CCA_KEYWORD_SIZE);
+        break;
+
+    default:
+        TRACE_ERROR("%s\n", ock_err(ERR_MECHANISM_PARAM_INVALID));
+        rc = CKR_MECHANISM_PARAM_INVALID;
+        goto done;
+    }
+
+    data_structure_length = 0;
+
+    dll_CSNDPKE(&return_code,
+                &reason_code,
+                NULL, NULL,
+                &rule_array_count,
+                rule_array,
+                (long *)&in_data_len,
+                in_data,
+                &data_structure_length,  // must be 0
+                NULL,           // ignored
+                (long *)&(attr->ulValueLen),
+                attr->pValue,
+                (long *)out_data_len,
+                out_data);
+
+    if (return_code != CCA_SUCCESS) {
+        TRACE_ERROR("CSNDPKE (RSA ENCRYPT) failed. return:%ld, reason:%ld\n",
+                    return_code, reason_code);
+        rc = CKR_FUNCTION_FAILED;
+        goto done;
+    } else if (reason_code != 0) {
+        TRACE_WARNING("CSNDPKE (RSA ENCRYPT) succeeded, but"
+                      " returned reason:%ld\n", reason_code);
```

Comment:
when I see this ...
the kernel coding guidelines say that strings shouldn't get wrapped but instead the 80 characters limit
is not enforced. Background is: if you grep the code for a string showing up in the traces you will not
find a wrapped string like this one

Reply:
Yea, but its that way all over the place......